test/integration/model/sync.test.js

```diff
@@ -403,6 +404,56 @@ describe(Support.getTestDialectTeaser('Model'), () => {
           expect(results.filter(r => r.name === 'wow_my_index')).to.have.length(1);
         }
       });
+
+      if (sequelize.dialect.supports.schemas) {
+        it('should not recreate a foreign key if it already exists when { alter: true } is used with a custom schema', async () => {
+          const schema = 'test';
+          await sequelize.createSchema(schema);
+
+          const User = sequelize.define('User', {}, { schema });
+          const BelongsToUser = sequelize.define('BelongsToUser', {}, { schema });
+          BelongsToUser.belongsTo(User, {  foreignKey: { targetKey: 'id', allowNull: false } });
+          await sequelize.sync({ alter: true });
+          await sequelize.sync({ alter: true });
+
+          const results = await sequelize
+            .getQueryInterface()
+            .getForeignKeyReferencesForTable(BelongsToUser.getTableName());
+          expect(results).to.have.length(1);
+          await sequelize.dropSchema(schema);
+        });
+
+        it('should not recreate a foreign key if it already exists when { alter: true } is used with a custom schema (reference attribute is a Model)', async () => {
+          const schema = 'test';
+          await sequelize.createSchema(schema);
+
+          class User extends Sequelize.Model {}
+
+          class BelongsToUser extends Sequelize.Model {}
+
+          User.init({
+            id: {
+              type: DataTypes.STRING,
+              primaryKey: true,
+            },
```

Comment:
is there any reason for making this a string, instead of the default auto-incremented integer id?